### PR TITLE
fix(ecs-patterns): `minHealthyPercent` and `maxHealthyPercent` props validation

### LINKED ISSUE #26158: aws-cdk-lib/aws-ecs-patterns: setting maxHealthyPercent incorrectly causes stack to fail to deploy
### Describe the bug

Setting the `maxHealthyPercent` on an `ApplicationLoadBalancedFargateService` to an invalid value does not raise a synth-time error, but it generates invalid CFN templates.

### Expected Behavior

Setting the `maxHealthyPercent` on an `ApplicationLoadBalancedFargateService` to a value such as 0.5 should trigger a compile-time error.

### Current Behavior

The invalid config is synthed without raising any warnings and results in the stack failing to apply.

### Reproduction Steps

Setting the `maxHealthyPercent` on an `ApplicationLoadBalancedFargateService` to a value such as 0.5.

### Possible Solution

Throw an exception at synth-time if `maxHealthyPercent` is not a whole integer between 0 and 100 inclusive.

### Additional Information/Context

_No response_

### CDK CLI Version

2.79.1

### Framework Version

_No response_

### Node.js Version

19.1.0

### OS

Linux

### Language

Typescript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by lpizzinidev**
Setting `maxHealthyPercent` to a non-integer value was not raising synth-time errors, but was generating invalid CFN templates.
This fix adds validation for both `maxHealthyPercent` and `minHealthyPercent`.

Closes #26158.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-ecs-patterns/lib/fargate/application-load-balanced-fargate-service.ts [hunk 1]
```diff
@@ -96,6 +96,9 @@ export class ApplicationLoadBalancedFargateService extends ApplicationLoadBalanc
       throw new Error('You must specify one of: taskDefinition or image');
     }
 
+    this.validateHealthyPercentage('minHealthyPercent', props.minHealthyPercent);
+    this.validateHealthyPercentage('maxHealthyPercent', props.maxHealthyPercent);
```
Comment (otaviomacedo) [inline packages/aws-cdk-lib/aws-ecs-patterns/lib/fargate/application-load-balanced-fargate-service.ts:100]: While we're here, please also validate that min < max.

**Comment by lpizzinidev** [comment]
Exemption request.
Let me know if an integration test is needed.

**Comment by lpizzinidev** [comment]
Exemption request.
This fix implements validation, it doesn't require an integration test.

**Comment by lpizzinidev** [comment]
@otaviomacedo 
This should be ready to merge

**Review by mrgrain — CHANGES_REQUESTED** [review]
Still a couple of issues. Assuming I understand correctly how this works and `min` should indeed by lower than `max`.

File: packages/aws-cdk-lib/aws-ecs-patterns/lib/fargate/application-load-balanced-fargate-service.ts [hunk 2]
```diff
@@ -96,6 +96,13 @@ export class ApplicationLoadBalancedFargateService extends ApplicationLoadBalanc
...
+
+    if (props.minHealthyPercent && props.maxHealthyPercent && props.minHealthyPercent >= props.maxHealthyPercent) {
```
[the remark this answers is not in the document]
Comment (mrgrain) [reply packages/aws-cdk-lib/aws-ecs-patterns/lib/fargate/application-load-balanced-fargate-service.ts:102]: Ah yes of course 🤦🏻  Just need to add the checks for unresolved tokens than please!

**Comment by mrgrain** [comment]
@lpizzinidev Sorry for that! We still need the checks for unresolved Tokens. Otherwise this is good to go.

**Review by mrgrain — APPROVED** [review]

**Pull request merged**